File: 2sem/lab5/lab5_sharp/NumericString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lab5_sharp
{
    class NumericString : BaseString
    {
        public NumericString(string str)  // constructor with parameter
        {
            String = str;
        }

        public float GetNumber()  // getting number in float
        {
            return Convert.ToSingle(String);
        }
    }
}

File: 2sem/lab5/lab5/NumericString.cpp
#include "NumericString.h"
#include <string>

NumericString::NumericString(std::string str) {  // constructor with parameter
	_string = str;
}

float NumericString::GetNumber() {  // getting number in float
	return std::stof(_string);
}

std::string NumericString::GetString() {  // getter of string
	return _string;
}